File: 백준/Bronze/2745. 진법 변환/진법 변환.py
N, B = input().split()
N = N[::-1]
num = '0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ'
sum = 0

for i in range(len(N)):
    for j in range(len(num)):
        if N[i] == num[j]:
            sum += j * (int(B)**i)
            break
print(sum)

File: 백준/Bronze/2745. 진법 변환/진법 변환.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        char[] n = scanner.next().toCharArray();
        int b = scanner.nextInt();
        int ans = 0;

        String num = "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        
        int power = 1;
        for (int i = 0; i < n.length; i++) {
            int value = num.indexOf(n[i]);
            ans *= b;
            ans += value;
        }

        System.out.println(ans);
    }
}
